### ISSUE TITLE
inno-updater.exe - Application Error (Windows)

### ISSUE THREAD

**Issue by wlamason** [Microsoft/vscode#57028]
Issue Type: <b>Bug</b>

When opening VS Code I am prompted to update to 'Visual Studio Code 1.26.1.'

VS Code states it is updating in the background.

'Restart Visual Studio Code to apply the latest update.'

No matter if I click 'Install Update' or restart VS Code manually, I am met with the same error every time:

inno-updater.exe - Application Error (dialog)
The application was unable to start correctly (0x000007b). Click OK to close the application.

I have read through many of the related issues, but haven't found anything that fixes this in particular. 

This occurred with the last version of VS Code I had installed, and eventually I got fed up and reinstalled with the new version.

Please let me know of any specific logs or information you would like me to provide to help resolve this.

Thanks!

VS Code version: Code 1.23.1 (d0182c3417d225529c6d5ad24b7572815d0de9ac, 2018-05-10T17:11:17.614Z)
OS version: Windows_NT x64 6.1.7601

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6600U CPU @ 2.60GHz (4 x 2808)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: disabled_software<br>video_decode: enabled<br>video_encode: enabled<br>vpx_decode: unavailable_software<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|15.88GB (10.07GB free)|
|Process Argv|C:\Program Files\Microsoft VS Code\Code.exe|
|Screen Reader|no|
|VM|0%|

</details>Extensions: none

**Comment by wlamason** [comment]
Hi Joao,

This is Windows 7 Enterprise.
![windows_7](https://user-images.githubusercontent.com/18597725/44585827-95275900-a77b-11e8-8e28-469a3027ae00.PNG)

Screenshot of the error:
![inno_updater_error](https://user-images.githubusercontent.com/18597725/44585845-a3757500-a77b-11e8-94c6-d64f3f5dbddc.PNG)

Running inno_updater.exe produces the same error as pictured above.

By "System Events to find the crash reason," are you referring to Windows Event Viewer? Like in this:
https://www.cyberlink.com/support/faq-content.do?id=10449

I am not familiar with using System Events. In the event viewer I am unable to find any errors or warnings with 0x000007b when filtering for critical, error, and warning on the event level. Let me know if this is what you are requesting, or something else.

Thank you for the reply!

**Comment by joaomoreno** [comment]
Lots of info here:

- https://stackoverflow.com/questions/10492037/the-application-was-unable-to-start-correctly-0xc000007b
- https://answers.microsoft.com/en-us/windows/forum/windows_10-performance-winpc/windows-10-the-application-was-unable-to-start/0589825b-7821-45e9-9e6c-d2b554bf17dc

Let me know if you find something.

**Comment by wlamason** [comment]
I went through the links you provided and did the following:

1. Ensured Microsoft Visual C++ was available in 32 and 64 bit versions.
1. Restarted computer.
1. Ensured .NET Framework is available on system. (.NET Framework 4.6.1 installed)
1. Ran the application as an administrator, and ran inno_updater.exe as an administrator.

None of the above proved fruitful. The dependency walk is displaying an error for inno_updater.exe stating: "Error: Modules with different CPU types were found."

![dependency_walk](https://user-images.githubusercontent.com/18597725/44596543-93b95900-a79a-11e8-9bd9-be7857efbdf0.PNG)

Went to Help > About in VS Code:
![vs_code_arch](https://user-images.githubusercontent.com/18597725/44596579-adf33700-a79a-11e8-806c-d3f8c9fe82bd.PNG)

I will have to spend some more time understanding the meaning of this, but wanted to provide an update.
